pop11_grammar_ebnf.txt

```diff
@@ -83,25 +87,25 @@ Jump ::=
 	'goto' Word
 
 Iterative ::=
-	'for' Expression 'step' Expression 'till' Expression 'do' Sequence 'endfor'
-	| 'while' Expression 'do' Sequence 'endwhile'
-	| 'until' Expression 'do' Sequence 'enduntil'
-	| 'for' Word (('in' | 'on') | ('from' Expression ('by' Expression)?)? 'to') Expression 'do' Sequence 'endfor'
-	| 'for' Word 'from_repeater' Expression 'do' Sequence 'endfor'
-	| 'foreach' Expression ('in' Expression)? 'do' Sequence 'endforeach'
-	| 'forevery' Expression ('in' Expression)? 'do' Sequence 'endforevery'
-	| 'repeat' (Expression 'times' | 'forever')? Sequence 'endrepeat'
+	'for' NonEmptyExpression 'step' NonEmptyExpression 'till' NonEmptyExpression 'do' Sequence 'endfor'
+	| 'while' NonEmptyExpression 'do' Sequence 'endwhile'
+	| 'until' NonEmptyExpression 'do' Sequence 'enduntil'
+	| 'for' Word (('in' | 'on') | ('from' NonEmptyExpression ('by' NonEmptyExpression)?)? 'to') NonEmptyExpression 'do' Sequence 'endfor'
+	| 'for' Word 'from_repeater' NonEmptyExpression 'do' Sequence 'endfor'

```

Comment:
`Word` should be `Variable`.

NonEmptyExpression is incorrect. It may be empty.
```
 Setpop
 : identfn(%termin%); for i from_repeater do i=> endfor;
 : 
```